pkg/types/types.go

```diff
@@ -105,6 +109,15 @@ type DNS struct {
 	Options     []string `json:"options,omitempty"`
 }
 
+type Attribute struct {
+	Name 	string
+	Value 	string
+}
+
+type CUSTOM struct {
+        CustomAttributes []Attribute
+}
+
```

Comment:
If the extra arguments/results is a plain dict, then we don't need these extra structures.